### PR TITLE
Renaming EventHandler, and adding utility function for reading without caring about overflow.

### PR DESCRIPTION
**Pull request by Rhuagh**
Also adding a read function that ignores overflow, and just logs an error message, and then returns the data iterator

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -78,6 +81,25 @@ where
     pub fn read(&self, reader_id: &mut ReaderId) -> Result<EventReadData<E>, EventError> {
         self.storage.read(reader_id)
     }
+
+    /// Read events and log an error event on overflow
+    pub fn read_ignore_overflow(
```
Comment (Xaeroxe) [inline src/lib.rs]: The name I had in mind was `lossy_read` which is fewer characters.

**Review by Xaeroxe — APPROVED** [review]
Looks great!  I just have one suggestion for naming.

File: src/lib.rs [hunk 2]
```diff
@@ -78,6 +81,25 @@ where
...
+        &self,
+        reader_id: &mut ReaderId,
+    ) -> Result<EventIterator<E>, EventError> {
+        match self.read(reader_id) {
+            Err(err) => Err(err),
```
Comment (torkleyy) [inline src/lib.rs]: Could use `self.read(reader_id).map`

Comment (Rhuagh) [reply src/lib.rs]: Ah, I keep forgetting about those, thanks! One of these days...

**Review by Xaeroxe — APPROVED** [review]
LGTM, thanks!

**Pull request merged**